ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,176 @@
+        objects.clear();
+        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, true) == 0);
+        REQUIRE(objects.size() == 1);
+    }
+
+    SECTION("Test delete functionality for Azure.", "[azure-connection]")
+    {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + object_name + "1");
+
+        blob_client.UploadFrom(create_file(object_name + "1", payload));
+
+        // Test that an object can be deleted.
+        REQUIRE(conn.delete_object(object_name + "1") == 0);
+
+        // Test that removing an object that doesn't exist returns a ENOENT.
+        REQUIRE(conn.delete_object(non_exist_object_key) == ENOENT);
+    }
+
+    SECTION("Check put functionality in Azure.", "[azure-connection]")
+    {
+        const std::string path = "./" + create_file(object_name, payload);
+
+        REQUIRE(conn.put_object(object_name + "1", path) == 0);
+
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + object_name + "1");
+
+        // Test that putting an object that doesn't exist locally returns -1.
+        REQUIRE(conn.put_object(non_exist_object_key, non_exist_object_key + ".txt") == -1);
+
+        blob_client.Delete();
+    }
+
+    SECTION("Check read functionality in Azure.", "[azure-connection]")
+    {
+        // Test reading whole file.
+        REQUIRE(conn.read_object(object_name, 0, payload.length(), buffer) == 0);
+        REQUIRE(static_cast<char *>(buffer) == payload);
+        memset(buffer, 0, 1024);
+
+        // Check that read works from the first ' ' to the end.
+        const int str_len = payload_2.length() - payload_2.find(" ");
+        REQUIRE(conn.read_object(object_name_2, payload_2.find(" "), str_len, buffer) == 0);
+        REQUIRE(static_cast<char *>(buffer) == payload_2.substr(payload_2.find(" "), str_len));
+        memset(buffer, 0, 1024);
+
+        // Test overflow on positive offset but past EOF.
+        REQUIRE(conn.read_object(object_name, 1, 1000, buffer) == -1);
+        memset(buffer, 0, 1024);
+
+        // Test overflow on negative offset but past EOF.
+        REQUIRE(conn.read_object(object_name, -1, 1000, buffer) == -1);
+        memset(buffer, 0, 1024);
+
+        // Test overflow with negative offset.
+        REQUIRE(conn.read_object(object_name, -1, 12, buffer) == -1);
+
+        // Tests that reading a existing object but wrong prefix returns a ENOENT.
+        REQUIRE(conn_bad.read_object(object_name, 0, 1, buffer) == ENOENT);
+
+        // Test that reading a non existent object returns a ENOENT.
+        REQUIRE(conn.read_object(non_exist_object_key, 0, 1, buffer) == ENOENT);
+    }
+
+    // Clean up.
+    for (auto pair : blob_objects) {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + pair.first);
+        blob_client.Delete();
+    }
+
+    // Sanity check that nothing exists.
+    free(buffer);
+    objects.clear();
+    REQUIRE(conn.list_objects(obj_prefix, objects, false) == 0);
```

Comment:
I would use the SDK to do this rather than the connection class.

Reply:
yup implemented